spec/unit/pops/types/iterable_spec.rb

```diff
@@ -0,0 +1,229 @@
+    not_iterable_types << PVariantType.new([iterable_types[0], not_iterable_types[0]])
+
+    iterable_types.each do |type|
+      it "should consider #{type} to be assignable to Iterable type" do
+        expect(PIterableType::DEFAULT.assignable?(type)).to eq(true)
+      end
+    end
+
+    not_iterable_types.each do |type|
+      it "should not consider #{type} to be assignable to Iterable type" do
+        expect(PIterableType::DEFAULT.assignable?(type)).to eq(false)
+      end
+    end
+  end
+
+  it 'does not wrap an Iterable in another Iterable' do
+    x = Iterable.on(5)
+    expect(Iterable.on(x)).to equal(x)
+  end
+
+  it 'produces a "times" iterable on integer' do
+    expect{ |b| Iterable.on(3).each(&b) }.to yield_successive_args(0,1,2)
+  end
+
+  it 'produces an iterable with element type Integer[0,X-1] for an iterable on an integer X' do
+    expect(Iterable.on(3).element_type).to eq(PIntegerType.new(0,2))
+  end
+
+  it 'produces a step iterable on an integer' do
+    expect{ |b| Iterable.on(8).step(3, &b) }.to yield_successive_args(0, 3, 6)
+  end
+
+  it 'produces a reverse iterable on an integer' do
+    expect{ |b| Iterable.on(5).reverse_each(&b) }.to yield_successive_args(4,3,2,1,0)
+  end
+
+  it 'produces an iterable on a integer range' do
+    expect{ |b| Iterable.on(2..7).each(&b) }.to yield_successive_args(2,3,4,5,6,7)
+  end
+
+  it 'produces an iterable with element type Integer[X,Y] for an iterable on an integer range (X..Y)' do
+    expect(Iterable.on(2..7).element_type).to eq(PIntegerType.new(2,7))
+  end
+
+  it 'produces an iterable on a character range' do
+    expect{ |b| Iterable.on('a'..'f').each(&b) }.to yield_successive_args('a', 'b', 'c', 'd', 'e', 'f')
+  end
+
+  it 'produces a step iterable on a range' do
+    expect{ |b| Iterable.on(1..5).step(2, &b) }.to yield_successive_args(1,3,5)
+  end
+
+  it 'produces a reverse iterable on a range' do
+    expect{ |b| Iterable.on(2..7).reverse_each(&b) }.to yield_successive_args(7,6,5,4,3,2)
+  end
+
+  it 'produces an iterable with element type String with a size constraint for an iterable on a character range' do
+    expect(Iterable.on('a'..'fe').element_type).to eq(PStringType.new(PIntegerType.new(1,2)))
+  end
+
+  it 'produces an iterable on a bounded Integer type' do
+    expect{ |b| Iterable.on(PIntegerType.new(2,7)).each(&b) }.to yield_successive_args(2,3,4,5,6,7)
+  end
+
+  it 'produces an iterable with element type Integer[X,Y] for an iterable on Integer[X,Y]' do
+    expect(Iterable.on(PIntegerType.new(2,7)).element_type).to eq(PIntegerType.new(2,7))
+  end
+
+  it 'produces an iterable on String' do
+    expect{ |b| Iterable.on('eat this').each(&b) }.to yield_successive_args('e', 'a', 't', ' ', 't', 'h', 'i', 's')
+  end
+
+  it 'produces an iterable with element type String[1,1] for an iterable created on a String' do
+    expect(Iterable.on('eat this').element_type).to eq(PStringType.new(PIntegerType.new(1,1)))
+ end
```

Comment:
so I was wrong earlier this example is correct; which means I misunderstood what the code was for...